evap/staff/tools.py

```diff
@@ -375,6 +379,30 @@ def remove_user_from_represented_and_ccing_users(user, ignored_users=None, test_
     return remove_messages
 
 
+def remove_inactive_participations(user, test_run=False):
+    remove_messages = []
+    evaluations = []
+    if user.is_active and user.can_be_marked_inactive_by_manager:
+        evaluations = user.evaluations_participating_in.filter(
+            state=Evaluation.State.PUBLISHED,
+            vote_end_date__lt=datetime.today() - timedelta(days=settings.PARTICIPATION_DELETION_AFTER_INACTIVE_MONTHS),
+        )
+    if len(evaluations) > 0:
+        if test_run:
+            remove_messages.append(
+                _("{} will be removed from {} participation(s) due to inactivity.").format(
+                    user.full_name, len(evaluations)
+                )
+            )
+        else:
+            for evaluation in evaluations:
+                evaluation.participants.remove(user)
+            remove_messages.append(
+                _("Removed {} from {} participation(s) due to inactivity.").format(user.full_name, len(evaluations))
+            )
```

Comment:
I don't think this is the correct logic for what we envisioned in #2176. We want:
> A user who hasn't participated in evaluations in the last X months is "long_inactive". Participations of "long_inactive" users are removed when performing bulk update

whereas this code currently does
> Given a user x, if x has any participations in evaluations that ended more than X months ago, remove them as a participant from those evaluations.

If I have been studying for 5 years and have been participating in evaluations regularly, I shouldn't be classified as "long_inactive" since my last evaluation is not that long ago. But I will still have participated in evaluations that have ended a long time ago.

---

There should be django queryset magic that allows to, given a set of users, annotate to each of them when exactly they last participated in an evaluation (proxied by the evaluation end date of the evaluations they participated in). Based on that distinction, we can then remove all participations of the user (or none). It's probably easier to explain this in person, so I recommend asking one of the maintainers/organizers at the next meetup :)

Reply:
Note that the test suite passed. This can be considered as an indication that the tests aren't assertive enough to distuingish correct from incorrect behavior, and should be expanded to include more edge-caseish situations (e.g. the one I outlined above).

Reply:
https://docs.djangoproject.com/en/5.1/topics/db/aggregation/#joins-and-aggregates

Reply:
Is this how it's supposed to be now?